lib/portage/sync/modules/git/git.py

```diff
@@ -228,6 +228,24 @@ def update(self):
         if not self.verify_head(revision="refs/remotes/%s" % remote_branch):
             return (1, False)
 
+        # Clean up the repo before trying to sync to upstream's
+        clean_cmd = [self.bin_command, "clean", "--force", "-d", "-x"]
```

Comment:
Let's add a ceiling on this so that it doesn't go above the repo dir as a sanity check in case e.g. /var is a git repo too or something.

Reply:
I've just pushed a commit that should do this, not tested it sufficiently yet. Will git really traverse to parent directories of a directory with a `.git` though? We should already only be operating on the directory if `.git` exists within it, via `GitSync.exists()`, called by `NewBase.sync()`.